### PR TITLE
[Xedra Evolved] I stole the inventor's sparks for dreamer class

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
Mods "[Xedra Evolved] I stole the inventor's sparks for dreamer class"
#### Describe the solution
Inventor got his own vitamin, that would work as his own mana in future (i hope) - Metaphysical Connection
800 of this vitamin (10 days) allow him to cast and obtain an item, with random magical properties (only active one) - they have few charges, but give pretty good advantage while using
Fixed the dreamer spells has lack of skill
Edited the inventor's spark spell to use much simpler EoC
For `light` monster, added him an ability to damage player also - it was intended to have one, but i was not smart enough to make one from the beginning. Now main strategy is cast and run
#### Additional context
I plan to replace spark mechanic with something smarter, that require the player to sit with laptop for sleepless hours to make some interesting items, but not now

### CODE REVIEW

File: data/mods/Xedra_Evolved/effects/effects.json [hunk 1]
```diff
@@ -112,6 +106,19 @@
     ],
     "rating": "good"
   },
+  {
+    "type": "effect_type",
+    "id": "dreamer_eff",
+    "name": [ "Matter Twist", "Matter Deformation", "Dimensional Shift", "Dimensional Chasm", "Reality Gateway" ],
+    "desc": [
+      "Sometimes the world nearby twist.  Was it you?",
+      "Sometimes there is a odd noise you hear.  Sound like whisper.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/effects/effects.json:115]:
```suggestion
      "Sometimes there is a odd noise you hear.  Sounds like a whisper.",
```

File: data/mods/Xedra_Evolved/effects/effects.json [hunk 2]
```diff
@@ -112,6 +106,19 @@
...
+      "You feel someone's presences.  They are wondering.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/effects/effects.json:116]: Are they wondering like thinking and you can kind of hear those thoughts or are they wandering around nearby?  Wanted to make sure I understand your intention on this.

File: data/mods/Xedra_Evolved/spells/XAEA.json [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+    "type": "SPELL",
+    "name": "Artifact Hologram Summon",
+    "description": "Summon a lot of holograms around you, for 0,5 - 1 h.",
+    "valid_targets": [ "ground" ],
+    "flags": [ "SILENT", "RANDOM_DAMAGE", "IGNORE_WALLS", "RANDOM_DURATION", "NO_EXPLOSION_SFX" ],
+    "effect": "summon",
+    "effect_str": "mon_hologram",
+    "shape": "blast",
+    "damage_type": "bash",
+    "message": "It is you.  And this is you.  And that is you.  You are everywhere.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/spells/XAEA.json:313]: I love this line a lot.

**Review by Maleclypse — COMMENTED** [review]
This is good stuff! Thank you!

File: data/mods/Xedra_Evolved/effects/effects.json (hunk 2 above)
Comment (GuardianDll) [reply data/mods/Xedra_Evolved/effects/effects.json:116]: More like "you feel they are interested in you, but still mostly neutral"

File: data/mods/Xedra_Evolved/effects/effects.json [hunk 4]
```diff
@@ -112,6 +106,19 @@
     ],
     "rating": "good"
   },
+  {
+    "type": "effect_type",
+    "id": "dreamer_eff",
+    "name": [ "Matter Twist", "Matter Deformation", "Dimensional Shift", "Dimensional Chasm", "Reality Gateway" ],
+    "desc": [
+      "Sometimes the world nearby twist.  Was it you?",
+      "Sometimes there is a odd noise you hear.  Sounds like a whisper.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/effects/effects.json:115]:
```suggestion
      "Sometimes there is an odd noise you hear.  Sounds like a whisper.",
```

File: data/mods/Xedra_Evolved/effects/effects.json [hunk 5]
```diff
@@ -112,6 +106,19 @@
     ],
     "rating": "good"
   },
+  {
+    "type": "effect_type",
+    "id": "dreamer_eff",
+    "name": [ "Matter Twist", "Matter Deformation", "Dimensional Shift", "Dimensional Chasm", "Reality Gateway" ],
+    "desc": [
+      "Sometimes the world nearby twist.  Was it you?",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/effects/effects.json:114]:
```suggestion
      "Sometimes the world nearby twists.  Was it you?",
```

File: data/mods/Xedra_Evolved/itemgroups/dreamer_artifacts.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
+[
+  {
+    "type": "item_group",
+    "//": "Inventor artifact creation itemgroup.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/itemgroups/dreamer_artifacts.json:4]:
```suggestion
    "//": "Dreamer artifact creation itemgroup.",
```

File: data/mods/Xedra_Evolved/spells/XAEA.json [hunk 7]
```diff
@@ -0,0 +1,389 @@
...
+  {
+    "id": "XAEA_teleport_rand",
+    "type": "SPELL",
+    "name": "Artifact Teleport",
+    "description": "Teleport the caster to a random point nearby.",
+    "effect": "short_range_teleport",
+    "shape": "blast",
+    "valid_targets": [ "ground" ],
+    "flags": [ "IGNORE_WALLS", "NO_LEGS", "SILENT", "RANDOM_TARGET" ],
+    "message": "It's not like you teleport somewhere else, rather the whole world has been shifted, and you just left in the same place.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/spells/XAEA.json:68]:
```suggestion
    "message": "It's not like you teleport somewhere else, rather the whole world has been shifted, and you are left in the same place.",
```

File: data/mods/Xedra_Evolved/spells/XAEA.json [hunk 8]
```diff
@@ -0,0 +1,389 @@
...
+    "name": { "str": "Artifact Heal" },
+    "description": "Will heal injuries all over your body.",
+    "valid_targets": [ "self" ],
+    "effect": "attack",
+    "flags": [ "NO_PROJECTILE", "NO_EXPLOSION_SFX", "SILENT" ],
+    "shape": "blast",
+    "base_casting_time": 100,
+    "min_damage": -30,
+    "max_damage": -30,
+    "message": "Your flesh start to rot and regenerate like in timelapse.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/spells/XAEA.json:104]:
```suggestion
    "message": "Your flesh quickly rots and regenerates like in timelapse.",
```

File: data/mods/Xedra_Evolved/spells/XAEA.json [hunk 9]
```diff
@@ -0,0 +1,389 @@
...
+  {
+    "type": "SPELL",
+    "id": "XAEA_mana",
+    "name": { "str": "Artifact Mana" },
+    "description": "Restore some mana.",
+    "min_damage": 1000,
+    "max_damage": 2000,
+    "valid_targets": [ "self" ],
+    "flags": [ "RANDOM_DAMAGE", "NO_EXPLOSION_SFX", "SILENT" ],
+    "message": "You feel your body feel energized, and your internal energy is restored.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/spells/XAEA.json:146]:
```suggestion
    "message": "You feel your body become energized, and you are ready to change the world.",
```

File: data/mods/Xedra_Evolved/spells/XAEA.json [hunk 10]
```diff
@@ -0,0 +1,389 @@
...
+    "max_duration": 720000,
+    "sound_description": "",
+    "message": ""
+  },
+  {
+    "id": "xe_int",
+    "type": "effect_type",
+    "name": [ "The Knowledge" ],
+    "desc": [ "You know for sure - right now you are the smartest human on this side of the planet." ],
+    "apply_message": "With some effort you can outsmart the reality.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/spells/XAEA.json:212]:
```suggestion
    "apply_message": "With some effort you can outsmart reality.",
```

File: data/mods/Xedra_Evolved/spells/XAEA.json [hunk 11]
```diff
@@ -0,0 +1,389 @@
...
+    "remove_message": "You feel yourself much less smarter than before.",
+    "rating": "good",
+    "show_intensity": false,
+    "enchantments": [ { "values": [ { "value": "INTELLIGENCE", "add": 12 } ] } ]
+  },
+  {
+    "id": "XAEA_electric_sight",
+    "type": "SPELL",
+    "name": "Artifact Electric Sight",
+    "description": "Give you an ability to see electric creatures for whole week.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/spells/XAEA.json:222]:
```suggestion
    "description": "Gives you an ability to see electric creatures for a whole week.",
```

**Review by Maleclypse — COMMENTED** [review]
I think this turned out really well.  Looked back over it and found a couple sentence structure changes but I don't see any changes needed in the mechanical parts.  Thanks!

**Review by Maleclypse — APPROVED** [review]
Thank you for the work you do!

**Pull request merged**